src/ImageSharp/Formats/Tiff/Writers/TiffPaletteWriter.cs

```diff
@@ -3,58 +3,47 @@
 
 using System;
 using System.Buffers;
-using System.IO;
 using System.Runtime.InteropServices;
-
-using SixLabors.ImageSharp.Compression.Zlib;
-using SixLabors.ImageSharp.Formats.Experimental.Tiff.Compression.Compressors;
-using SixLabors.ImageSharp.Formats.Experimental.Tiff.Utils;
-using SixLabors.ImageSharp.Formats.Tiff.Compression;
+using SixLabors.ImageSharp.Advanced;
+using SixLabors.ImageSharp.Formats.Experimental.Tiff.Compression;
 using SixLabors.ImageSharp.Memory;
 using SixLabors.ImageSharp.Metadata.Profiles.Exif;
 using SixLabors.ImageSharp.PixelFormats;
 using SixLabors.ImageSharp.Processing.Processors.Quantization;
 
 namespace SixLabors.ImageSharp.Formats.Experimental.Tiff.Writers
 {
-    /// <summary>
-    /// Utility class for writing TIFF data to a <see cref="Stream"/>.
-    /// </summary>
-    internal class TiffPaletteWriter : TiffBaseColorWriter
+    internal class TiffPaletteWriter<TPixel> : TiffBaseColorWriter<TPixel>
+        where TPixel : unmanaged, IPixel<TPixel>
     {
-        /// <summary>
-        /// Initializes a new instance of the <see cref="TiffPaletteWriter" /> class.
-        /// </summary>
-        /// <param name="output">The output stream.</param>
-        /// <param name="memoryAllocator">The memory allocator.</param>
-        /// <param name="configuration">The configuration.</param>
-        /// <param name="entriesCollector">The entries collector.</param>
-        public TiffPaletteWriter(TiffStreamWriter output, MemoryAllocator memoryAllocator, Configuration configuration, TiffEncoderEntriesCollector entriesCollector)
-             : base(output, memoryAllocator, configuration, entriesCollector)
+        private const int ColorsPerChannel = 256;
+        private const int ColorPaletteSize = ColorsPerChannel * 3;
+        private const int ColorPaletteBytes = ColorPaletteSize * 2;
+
+        private readonly IndexedImageFrame<TPixel> quantized;
+
+        public TiffPaletteWriter(ImageFrame<TPixel> image, IQuantizer quantizer, MemoryAllocator memoryAllocator, Configuration configuration, TiffEncoderEntriesCollector entriesCollector)
+             : base(image, memoryAllocator, configuration, entriesCollector)
         {
+            using IQuantizer<TPixel> frameQuantizer = quantizer.CreatePixelSpecificQuantizer<TPixel>(this.Configuration);
+            this.quantized = frameQuantizer.BuildPaletteAndQuantizeFrame(image, image.Bounds());
```

Comment:
`Palette` encoder also use hard operations

Reply:
This should utilize whatever bit depth property we store.

Reply:
@JimBobSquarePants Sorry, I don't understand, what did you mean?

Reply:
I'm seeing a passed `IQuantizer` instance but also hardcoded `ColorsPerChannel` values which indicate 256 colors each time. I would assume that there is some sort of `BitDepth` property that is captured during decoding and can be optionally set to dictate the encoding bit depth. There should be consistency in the palette size and desired bit depth.

Reply:
Yeah, it's required to bit depth settings, but at the moment support only 8 bit palette and grayscale.
I would like add bit depth later, in future PR.